apps/emqx/test/emqx_persistent_session_SUITE.erl

```diff
@@ -444,6 +372,7 @@ t_assigned_clientid_persistent_session(Config) ->
     {ok, Client2} = emqtt:start_link([
         {clientid, AssignedClientId},
         {proto_ver, v5},
+        {properties, #{'Session-Expiry-Interval' => 30}},
```

Comment:
Discussion point. @thalesmg @ieQu1 
Moderately stupid consequence of how the session implementation [is now chosen](https://github.com/emqx/emqx/blob/3e679df1b6b1bf7e8d8769a7c13734ad57481a24/apps/emqx/src/emqx_session.erl#L557-L565) is that this line is now required for this testcase to pass under `emqx_persistent_session_ds` implementation. Oddly enough, the spec does not explicitly say what the Client wants when it says CleanStart = 0 **and** SessionExpiryInterval = 0 (which is the default), yet after one more read-through I have an impression that the spec wants us to continue with existing session, the `emqx_persistent_session_ds` in this case, meaning that the testcase should pass w/o this added line. Though it's still not clear what SessionExpiryInterval = 0 means then. I wonder how would you interpret the spec?

Reply:
Sounds reasonable, thanks!

Still have some doubts. The [MQTT-3.1.2-5] and this piece from the spec

> If it is set to 0, or is absent, the Session ends when the Network Connection is closed. 

...Kinda leave the impression that (if read literally) we should take the existing session, but update its expiry interval to 0. So that it will perish once the _new_ client connection goes away.

<sup>On a sidenote: perhaps it should be fine to decide which interpretation to assume in a followup PR. 🤔 </sup>

Reply:
I agree, if the original connection had `SessionExpiryInterval > 0`, those items suggest the behavior you described.  

If the _original connection_ had `SessionExpiryInterval = 0`, then I think even with `CleanStart = 0` we should destroy the session rather than letting it take over.

Reply:
> I agree, if the original connection had `SessionExpiryInterval > 0`, those items suggest the behavior you described.

👍🏼 

This is exactly what's happening in the testcase. Which then means we need to look in persistent session storage first for any CONNECT with `CleanStart = 0` (we're kinda lucky there's only one persistent session storage currently 🌝).

I'll try to tackle that in the current PR, hope it won't require too much effort.

> If the _original connection_ had `SessionExpiryInterval = 0`, then I think even with `CleanStart = 0` we should destroy the session rather than letting it take over.

Agree. It should have already been destroyed in fact, by the time the next connection is established.

Reply:
Made an attempt to revamp this in 05b304ebe.